File: web/src/types/dailyDisciplineReport.ts
import type { SourceHealthItem } from './market'
import type { RiskAlert } from './riskAlert'

export type DailyDisciplineReportStatus =
  | 'not_started'
  | 'running'
  | 'success'
  | 'degraded'
  | 'failed'
  | 'insufficient_data'

export interface DailyDisciplineReportEvidenceSummary {
  evidence_count: number
  independent_source_count: number
  high_grade_independent_source_count: number
}

export interface DailyDisciplineReportTrendSummary {
  success_count: number
  degraded_count: number
  failed_count: number
  insufficient_data_count: number
}

export interface DailyDisciplineReportP34Coverage {
  summary?: string
  missing_categories?: string[]
  source_health?: SourceHealthItem[]
}

export interface DailyDisciplineReport {
  report_id: string
  local_date: string
  scope: string
  status: DailyDisciplineReportStatus
  summary: string
  source_type?: string
  source_id?: string
  decision_id?: string
  decision_link?: string
  auto_run_link?: string
  audit_link?: string
  notification_link?: string
  failure_code?: string
  failure_reason?: string
  missing_action?: string
  missing_categories?: string[]
  final_verdict?: string
  verdict_status?: string
  evidence: DailyDisciplineReportEvidenceSummary
  p34_source_coverage?: DailyDisciplineReportP34Coverage
  risk_alerts?: RiskAlert[]
  trend: DailyDisciplineReportTrendSummary
  safety_note: string
  updated_at?: string
}

export interface DailyDisciplineReportListResponse {
  reports: DailyDisciplineReport[]
}

File: internal/application/dto/daily_discipline_report.go
package dto

// DailyDisciplineReportResponse exposes the local daily discipline report for UI review.
type DailyDisciplineReportResponse struct {
	ReportID          string                           `json:"report_id"`
	LocalDate         string                           `json:"local_date"`
	Scope             string                           `json:"scope"`
	Status            string                           `json:"status"`
	Summary           string                           `json:"summary"`
	SourceType        string                           `json:"source_type"`
	SourceID          string                           `json:"source_id"`
	DecisionID        string                           `json:"decision_id"`
	DecisionLink      string                           `json:"decision_link"`
	AutoRunLink       string                           `json:"auto_run_link"`
	AuditLink         string                           `json:"audit_link"`
	NotificationLink  string                           `json:"notification_link"`
	FailureCode       string                           `json:"failure_code"`
	FailureReason     string                           `json:"failure_reason"`
	MissingAction     string                           `json:"missing_action"`
	MissingCategories []string                         `json:"missing_categories"`
	FinalVerdict      string                           `json:"final_verdict"`
	VerdictStatus     string                           `json:"verdict_status"`
	Evidence          DailyDisciplineReportEvidence    `json:"evidence"`
	P34SourceCoverage DailyDisciplineReportP34Coverage `json:"p34_source_coverage"`
	RiskAlerts        []DailyDisciplineReportRiskAlert `json:"risk_alerts"`
	Trend             DailyDisciplineReportTrend       `json:"trend"`
	SafetyNote        string                           `json:"safety_note"`
	UpdatedAt         string                           `json:"updated_at"`
}

// DailyDisciplineReportEvidence summarizes evidence coverage for the linked decision.
type DailyDisciplineReportEvidence struct {
	EvidenceCount                   int `json:"evidence_count"`
	IndependentSourceCount          int `json:"independent_source_count"`
	HighGradeIndependentSourceCount int `json:"high_grade_independent_source_count"`
}

// DailyDisciplineReportP34Coverage summarizes expanded public data coverage used by the report.
type DailyDisciplineReportP34Coverage struct {
	Summary           string             `json:"summary"`
	MissingCategories []string           `json:"missing_categories"`
	SourceHealth      []SourceHealthItem `json:"source_health"`
}

// DailyDisciplineReportTrend summarizes recent report statuses.
type DailyDisciplineReportTrend struct {
	SuccessCount          int `json:"success_count"`
	DegradedCount         int `json:"degraded_count"`
	FailedCount           int `json:"failed_count"`
	InsufficientDataCount int `json:"insufficient_data_count"`
}

// DailyDisciplineReportRiskAlert summarizes a related P35 risk alert for report surfaces.
type DailyDisciplineReportRiskAlert struct {
	AlertID           string   `json:"alert_id"`
	RiskType          string   `json:"risk_type"`
	Severity          string   `json:"severity"`
	SOPStatus         string   `json:"sop_status"`
	Symbol            string   `json:"symbol"`
	TriggerSummary    string   `json:"trigger_summary"`
	ProhibitedActions []string `json:"prohibited_actions"`
	SuggestedActions  []string `json:"suggested_actions"`
	Link              string   `json:"link"`
	SafetyNote        string   `json:"safety_note"`
}

// DailyDisciplineReportListResponse wraps daily discipline report list results.
type DailyDisciplineReportListResponse struct {
	Reports []DailyDisciplineReportResponse `json:"reports"`
}
